ctru-rs/src/lib.rs

```diff
@@ -11,14 +11,27 @@ pub fn init() {
 
     use std::panic::PanicInfo;
 
+    let main_thread = thread::current().id();
+
     // Panic Hook setup
     let default_hook = std::panic::take_hook();
     let new_hook = Box::new(move |info: &PanicInfo| {
-        println!("\x1b[1;31m\n--------------------------------------------------");
+        println!("\x1b[1;31m\n--------------------------------------------------"); // Red ANSI Color Code
         default_hook(info);
-        println!("\nThe thread will exit in 5 seconds");
 
-        thread::sleep(std::time::Duration::from_secs(5));
+        // Only for panics in the main thread
+        if main_thread == thread::current().id() {
+            println!("\nPress SELECT to exit the software");
+            let hid = services::hid::Hid::init().unwrap();
+
+            loop {
+                hid.scan_input();
+                if hid.keys_down().contains(services::hid::KeyPad::KEY_SELECT) {
+                    break;
+                }
+            }
+        }
+        println!("\x1b[1;37m\n"); // Get back to white
```

Comment:
Would it be a little more agnostic to use a "reset" ansi code (either `1;39` or `0`) instead of explicitly white? I haven't tested the behavior of that vs using colors...

Also there's an extra `\n` here, not sure if that's intentional

```suggestion
        println!("\x1b[0m"); // Get back to white
```

Reply:
There's a decent crate [`ansi_term`](https://docs.rs/ansi_term/latest/ansi_term/) that does all the work of remembering what codes mean and when to reset styles, etc. which is decent. I was also experimenting with [`ansi-escapes`](https://docs.rs/ansi-escapes/latest/ansi_escapes/) for cursor positioning. IIRC both of them built and worked as expected on 3DS, but it might not be desirable to add them as deps of `ctru-rs` itself

Reply:
Generally speaking it will be the user to print those, so it's at their discretion. This are the only `print!`s in the crate, we don't need to add dependencies for now.